### PR TITLE
Add RPMB simulation support on host side

### PR DESCRIPTION
**Pull request by James-Zzz**
rpmb_dev application listens to socket which connected to QEMU guest,
it handles data sent from QEMU guest, and operate RPMB_DATA on demand.
New virtio serial device would be exposed to QEMU guest, RPMB service
communicated with this virtio port for event handling.

Tracked-On: OAM-91232
Signed-off-by: Zhong,Fangjian <fangjian.zhong@intel.com>

### CODE REVIEW

File: groups/device-specific/caas/start_android_qcow2.sh [hunk 1]
```diff
@@ -393,13 +404,39 @@ function setup_vsock_host_utilities(){
...
 function cleanup_vsock_host_utilities(){
 	if [[ $enable_vsock == "true" ]]; then
 		kill -9 `pidof batsys`
 		kill -9 `pidof thermsys`
 	fi
 }
 
+function cleanup_rpmb_sock(){
+	# Send kill signal to rpmb_dev, remove RPMB sock
+	kill -9 `pidof rpmb_dev`
```
Comment (YadongQi) [inline groups/device-specific/caas/start_android_qcow2.sh:436]: What if we have several CiV instances? you will kill all backend process?

Comment (James-Zzz) [reply groups/device-specific/caas/start_android_qcow2.sh:436]: Yes, you are right.
If we need to support multiple CIV instance simultaneous, we need to this clean up logic, not just for rpmb_sock, but also for vsock.

**Comment by James-Zzz** [comment]
Commit message updated.

File: groups/device-specific/caas/start_android_qcow2.sh (hunk 1 above)
Comment (James-Zzz) [reply groups/device-specific/caas/start_android_qcow2.sh]: This issue fixed, we will kill child rpmb_dev process created by current process only.

File: groups/device-specific/caas/start_android_qcow2.sh [hunk 2]
```diff
@@ -22,6 +22,13 @@ GVTg_VGPU_TYPE="i915-GVTg_V5_4"
 SDONLY_DEV_PATH="/dev/mmcblk0p1"
 SDEMMC_DEV_PATH="/dev/mmcblk1p1"
 
+RPMB_PATH=${scripts_dir}
```
Comment (YadongQi) [inline groups/device-specific/caas/start_android_qcow2.sh:25]: these are not script related. how about  RPBM_PATH=${work_dir}/rpmb ?

File: groups/device-specific/caas/start_android_qcow2.sh [hunk 3]
```diff
@@ -393,13 +405,42 @@ function setup_vsock_host_utilities(){
...
+
+	# RPMB sock should be removed at cleanup, if there exists RPMB sock,
+	# rpmb_dev cannot be launched succefully. Delete any exist RPMB sock,
+	# rpmb_dev application creates RPMB sock by itself.
+	if [ -S ${RPMB_SOCK} ]; then
+		rm ${RPMB_SOCK}
+	fi
+
+	# rmpb_dev should be launched in other thread
+	${RPMB_DEV} --dev ${RPMB_DATA} --sock ${RPMB_SOCK} &
```
Comment (YadongQi) [inline groups/device-specific/caas/start_android_qcow2.sh:425]: what if this process failed to start?

File: groups/device-specific/caas/start_android_qcow2.sh [hunk 4]
```diff
@@ -393,13 +405,42 @@ function setup_vsock_host_utilities(){
...
+	fi
+
+	# RPMB sock should be removed at cleanup, if there exists RPMB sock,
+	# rpmb_dev cannot be launched succefully. Delete any exist RPMB sock,
+	# rpmb_dev application creates RPMB sock by itself.
+	if [ -S ${RPMB_SOCK} ]; then
+		rm ${RPMB_SOCK}
+	fi
+
+	# rmpb_dev should be launched in other thread
```
Comment (YadongQi) [inline groups/device-specific/caas/start_android_qcow2.sh:424]: other process?

Comment (James-Zzz) [reply groups/device-specific/caas/start_android_qcow2.sh:424]: fixed

File: groups/device-specific/caas/start_android_qcow2.sh (hunk 3 above)
Comment (James-Zzz) [reply groups/device-specific/caas/start_android_qcow2.sh:425]: Suggest to add this feature in next patch which used to monitor all child process created by this script.

File: groups/device-specific/caas/start_android_qcow2.sh (hunk 2 above)
Comment (James-Zzz) [reply groups/device-specific/caas/start_android_qcow2.sh:25]: As discussed, keep rpmb_dev application in script folder, move RPMB_DATA and rpmb_soc to work dir.

**Pull request merged**